test/functional/framework.go

```diff
@@ -169,6 +189,26 @@ func (f *FluentdFunctionalFramework) Deploy() (err error) {
 	if err != nil {
 		return fmt.Errorf("service could not be started")
 	}
+
+	log.V(2).Info("waiting for fluentd to be ready")
+	err = wait.PollImmediate(time.Second*2, time.Second*30, func() (bool, error) {
```

Comment:
test/client/Client.Watch is a more efficient way to wait for pods but there is a lot of code still using polling so I wouldn't hold up this merge just for that. At some point we'll need to make a sweep of the tests to bring them up to date